apps/emqx_authz/test/emqx_authz_http_test_server.erl

```diff
@@ -0,0 +1,91 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2020-2021 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_authz_http_test_server).
+
+-behaviour(supervisor).
+-behaviour(cowboy_handler).
+
+% cowboy_server callbacks
+-export([init/2]).
+
+% supervisor callbacks
+-export([init/1]).
+
+% API
+-export([start_link/2,
+         stop/0,
+         set_handler/1
+        ]).
+
+%%------------------------------------------------------------------------------
+%% API
+%%------------------------------------------------------------------------------
+
+start_link(Port, Path) ->
+    supervisor:start_link({local, ?MODULE}, ?MODULE, [Port, Path]).
+
+stop() ->
+    Sup = ?MODULE,
+    [{Id, _Child, _Type, _Modules}] = supervisor:which_children(Sup),
+    ok = supervisor:terminate_child(Sup, Id),
```

Comment:
Q: is it really needed to delete the children explicitly like this? `monitor(SupPid), exit(SupPid, shutdown), receive {'DOWN', ...`  should do the trick?

Reply:
I simplified this part a bit more, we can just terminate `Sup`, without deleting children expilicitly.

But seems I don't get the idea about monitor.
`stop` is used for graceful shutdown of the test server, e.g.

```erlang
{ok, _} = emqx_authz_http_test_server:start_link(33333, "/foo"),
...test smth...
ok = emqx_authz_http_test_server:stop(),

{ok, _} = emqx_authz_http_test_server:start_link(44444, "/bar"),
...test smth else...
ok = emqx_authz_http_test_server:stop().
```

If we forget to stop in tests, or if a error ocurrs, then the test http server is also shut down because now it is linked.

Reply:
Ah, got it :)
But now with `gen_server:stop` we do not need this, I think.